utilities/hco.py

```
@@ -68,6 +71,43 @@
+    if not v1_version:
+        raise ValueError("HCO CRD does not have a v1 version")
+
+    fg_description = v1_version["schema"]["openAPIV3Schema"]["properties"]["spec"]["properties"]["featureGates"].get(
+        "description", ""
+    )
+    phases = dict(_FG_PHASE_RE.findall(fg_description))
+    if not phases:
+        raise ValueError(
+            f"Failed to parse FG phases from HCO CRD — format may have changed. "
+            f"Description starts with: {fg_description[:200]!r}"
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)